pkg/ctl/upgrade/cluster.go

```diff
@@ -142,3 +127,70 @@ func DoUpgradeCluster(cmd *cmdutils.Cmd) error {
 
 	return nil
 }
+
+func requiresVersionUpgrade(clusterMeta *api.ClusterMeta, currentEKSVersion string) (bool, error) {
+	nextVersion, err := getNextVersion(currentEKSVersion)
+	if err != nil {
+		return false, err
+	}
+
+	// If the version was not specified default to the next Kubernetes version and assume the user intended to upgrade if possible
+	if clusterMeta.Version == "" {
+		if api.IsSupportedVersion(nextVersion) {
+			clusterMeta.Version = nextVersion
+			return true, nil
+		}
+
+		// There is no new version, stay in the current one
+		clusterMeta.Version = currentEKSVersion
+		return false, nil
+	}
+
+	if c, err := utils.CompareVersions(clusterMeta.Version, currentEKSVersion); err != nil {
+		return false, err
+	} else if c < 0 {
+		return false, fmt.Errorf("cannot upgrade to a lower version. Found given target version %q, current cluster version %q", clusterMeta.Version, currentEKSVersion)
+	}
+
+	if api.IsDeprecatedVersion(clusterMeta.Version) {
+		return false, fmt.Errorf("control plane version %q has been deprecated", clusterMeta.Version)
+	}
+
+	if !api.IsSupportedVersion(clusterMeta.Version) {
+		return false, fmt.Errorf("control plane version %q is not known to this version of eksctl, try to upgrade eksctl first", clusterMeta.Version)
+	}
+
+	if clusterMeta.Version == currentEKSVersion {
+		return false, nil
+	}
+
+	if clusterMeta.Version == nextVersion {
+		return true, nil
+	}
+
+	return false, fmt.Errorf(
+		"upgrading more than one version at a time is not supported. Found upgrade from %q to %q. Please upgrade to %q first",
+		currentEKSVersion,
+		clusterMeta.Version,
+		nextVersion)
+}
+
+func getNextVersion(currentVersion string) (string, error) {
+	switch currentVersion {
+	case "":
+		return "", errors.New("unable to get control plane version")
+	case api.Version1_12:
+		return api.Version1_13, nil
+	case api.Version1_13:
+		return api.Version1_14, nil
+	case api.Version1_14:
+		return api.Version1_15, nil
+	case api.Version1_15:
+		return api.Version1_16, nil
+	case api.Version1_16:
+		return "1.17", nil
```

Comment:
1.17 is not supported yet, maybe add a TODO for when 1.17 is released?

Reply:
I had to change how this function works and return the next version even if it's not supported. Then it will be caught later in https://github.com/weaveworks/eksctl/pull/2318/files#diff-a842b70d6bec5fa853663713afacdbfaR114

Reply:
I made this test more obvious now https://github.com/weaveworks/eksctl/pull/2318/files#diff-105010fad99ed058ae4cf85133eca859R222

Reply:
nitpick: might still be worth adding the constant `api.Version1_17` here